File: 2023/advent_of_code/test/q25/q25_test.exs
defmodule AdventOfCode.Q25Test do
  use ExUnit.Case
  doctest AdventOfCode.Q25

  test "Q25 Part 1 Example works" do
    file_stream = File.stream!("./test/q25/q25_example.txt")
    result = AdventOfCode.Q25.part1(file_stream)
    assert result == 54
  end

  test "Q25 Part 1 works" do
    file_stream = File.stream!("./test/q25/q25.txt")
    result = AdventOfCode.Q25.part1(file_stream)
    assert result == 567_606
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q25/Q25Test.kt
package q25

import junit.framework.TestCase.assertEquals
import java.nio.file.Paths
import kotlin.test.Test

class Q25Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q25.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals("2=--00--0220-0-21==1", part1(filePath!!))
    }
}